### PR TITLE
Create 2018-05-30-introducing-bazel-cquery.md

### PR DESCRIPTION
**Pull request by juliexxia**
Attempt #3 to public cquery blogpost. Jin adding you in here because I feel like you caught an issue last time!

### CODE REVIEW

File: _posts/2018-05-29-Introducing-Bazel-Cquery.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+What’s the difference? `query` doesn’t understand build flags and returns all possible answers to a given query expression. `cquery` (configurable query) runs at a later point in the build process, after flag evaluation and [configurable attribute](https://docs.bazel.build/versions/master/be/common-definitions.html#configurable-attributes) resolution. Thus, it can understand [build options](https://docs.bazel.build/versions/master/command-line-reference.html#command-line-reference) and give the answer to a specific Bazel invocation as dictated by its set of flags. Since it runs at a later point, ‘cquery’ is by nature slower than ‘query’.  
+
+`cquery` has yet to reach feature parity with `query`, but as we develop it further, `cquery` will also be able to expose information to which `query` does not have access. See the comparison below for what’s currently supported. 
+
+|                      | `query`                                                                    |`cquery`                              | both                                                                            |
```
Comment (jin) [inline _posts/2018-05-29-Introducing-Bazel-Cquery.md]: Can you please use a HTML table instead? Feel free to reuse the code in the previous PR: https://github.com/bazelbuild/bazel-blog/pull/81

File: _posts/2018-05-29-Introducing-Bazel-Cquery.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+Bazel uses the [Bazel Query Language](https://docs.bazel.build/versions/master/query.html) to articulate questions such as those listed above. This language includes a set of [functions](https://docs.bazel.build/versions/master/query.html), such as ‘deps’ and ‘somepath’, which can be combined and nested to describe more complex queries. 
+
+The query commands also support a set of flag-based options that govern query behavior. For example, [--noimplicit_deps](https://docs.bazel.build/versions/master/query.html#implicit_deps) limits results to only include targets that are explicitly declared in a BUILD file. The flags also support a selection of [output formats](https://docs.bazel.build/versions/master/query.html#output-formats) for the results of the query, such as a [graph output](https://docs.bazel.build/versions/master/query.html#output-graph) (especially helpful for visualizing the collection of paths between two nodes).
+
+## Introducing `bazel cquery`
+Bazel has long supported the `query` command. We’re excited to announce a second* command, `cquery`!
```
Comment (jin) [inline _posts/2018-05-29-Introducing-Bazel-Cquery.md]: escape `second\*`

File: _posts/2018-05-29-Introducing-Bazel-Cquery.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+What’s the difference? `query` doesn’t understand build flags and returns all possible answers to a given query expression. `cquery` (configurable query) runs at a later point in the build process, after flag evaluation and [configurable attribute](https://docs.bazel.build/versions/master/be/common-definitions.html#configurable-attributes) resolution. Thus, it can understand [build options](https://docs.bazel.build/versions/master/command-line-reference.html#command-line-reference) and give the answer to a specific Bazel invocation as dictated by its set of flags. Since it runs at a later point, ‘cquery’ is by nature slower than ‘query’.
```
Comment (jin) [inline _posts/2018-05-29-Introducing-Bazel-Cquery.md]: standardize single quotes to backticks around `cquery` and `query`?

**Comment by jin** [comment]
Ah I see you've been added to config.yml, feel free to skip that :-)

**Comment by jmmv** [comment]
Note that I also moved the original posts into `_drafts` while fixing the authorship issue. Please change this PR to *move* that file from the `_drafts` folder into `_posts`, and to apply content changes appropriately.

File: _posts/2018-05-29-Introducing-Bazel-Cquery.md (hunk 1 above)
Comment (jmmv) [reply _posts/2018-05-29-Introducing-Bazel-Cquery.md]: Why? I fixed our setup to also parse tables, and added some templating to make them look nicer. See https://blog.bazel.build/2018/04/13/preliminary-sandboxfs-support.html for a rendered example.

File: _posts/2018-05-29-Introducing-Bazel-Cquery.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    values = {"define": "species=excelsior"},
+)
+config_setting(
+    name = "americana",
+    values = {"define": "species=americana"},
+)
+
+# Traditional query
+$ bazel query "deps(//tree:ash)" --define species=excelsior
+...error because query does not understand --define…
```
Comment (jmmv) [inline _posts/2018-05-29-Introducing-Bazel-Cquery.md]: Be aware that your editor "helpfully" replaced the trailing ... here with an ellipsis character (but not the first one), which is probably not what you want.

File: _posts/2018-05-29-Introducing-Bazel-Cquery.md (hunk 1 above)
Comment (jin) [reply _posts/2018-05-29-Introducing-Bazel-Cquery.md]: Oh nice! I didn't realize you fixed it already.

**Comment by jin** [comment]
https://blog.bazel.build/2018/06/06/introducing-bazel-cquery.html

**Pull request closed without merging**